src/rgw/rgw_rados.cc

```diff
@@ -6794,6 +6794,22 @@ int RGWRados::block_while_resharding(RGWRados::BucketShard *bs,
 	ldpp_dout(dpp, 10) << __func__ <<
 	  " INFO: was able to take reshard lock for bucket " <<
 	  bucket_id << dendl;
+        ret = fetch_new_bucket_info("trying_to_clear_resharding");
+        if (ret < 0) {
+	  reshard_lock.unlock();
+	  ldpp_dout(dpp, 0) << __func__ <<
+	    " ERROR: failed to update bucket info before clear resharding for bucket " <<
+	    bucket_id << dendl;
+          continue; // try again
+        }
+        if (!bucket_info.layout.target_index) {
+          // there is nothing to revert
```

Comment:
i believe we still want to call `RGWBucketReshard::clear_resharding()` -> `cancel_reshard()` -> `set_resharding_status(NOT_RESHARDING)`, since we saw that `entry.resharding_in_progress()` was true above

Reply:
so, should we make check `bucket_info.layout.target_index` inside `cancel_reshard()` instead?

Reply:
yeah, that sounds right

Reply:
moved the check to here: https://github.com/ceph/ceph/pull/44253/files#diff-79d1c1fa1e35ac24a2407eab52f02ab12cab04370c3956e38e725d9d6d6b46f3R439